suite-native/discovery/src/discoveryMiddleware.ts

```diff
@@ -55,6 +65,29 @@ export const prepareDiscoveryMiddleware = createMiddlewareWithExtraDeps(
             );
         }
 
+        // this is for further continual rerunning discovery for various other reasons
+        if (
+            selectDeviceThunk.fulfilled.match(action) ||
+            setEnabledDiscoveryNetworkSymbols.match(action) ||
+            toggleEnabledDiscoveryNetworkSymbols.match(action) ||
+            accountsActions.changeAccountVisibility.match(action) ||
+            deviceActions.updateSelectedDevice.match(action)
+        ) {
+            // check whether we consider some network not to be fully discovered
+            const shouldRunDiscoveryForDevice = selectShouldRunDiscoveryForDevice(getState());
+            // check whether we are allowed to run the discovery now
+            const canRunDiscoveryForDevice = selectCanRunDiscoveryForDevice(getState());
+
+            if (device?.state && canRunDiscoveryForDevice && shouldRunDiscoveryForDevice) {
```

Comment:
`device?.state && ` seems to be redundant as part of canRunDiscoveryForDevice
```suggestion
            if (canRunDiscoveryForDevice && shouldRunDiscoveryForDevice) {
```

Reply:
b675fc3e18